baselines/dom.generated.d.ts

```diff
@@ -19886,10 +19886,10 @@ declare var onsuspend: ((this: Window, ev: Event) => any) | null;
  */
 declare var ontimeupdate: ((this: Window, ev: Event) => any) | null;
 declare var ontoggle: ((this: Window, ev: Event) => any) | null;
-declare var ontouchcancel: ((this: Window, ev: TouchEvent) => any) | null;
-declare var ontouchend: ((this: Window, ev: TouchEvent) => any) | null;
-declare var ontouchmove: ((this: Window, ev: TouchEvent) => any) | null;
-declare var ontouchstart: ((this: Window, ev: TouchEvent) => any) | null;
+declare var ontouchcancel?: ((this: Window, ev: TouchEvent) => any) | null;
```

Comment:
Ah, no, this won't work.... Sorry to make you break code 🙄

Reply:
Should I let this PR lapse until such work is complete, or return to the original solution that will break if the auto-generated types ever change?

Reply:
Okay, I got a working way. Please fix `src/emitter.ts` line 620 as:

```diff
-            const requiredModifier = p.required === undefined || p.required === 1 ? "" : "?";
+            const requiredModifier = p.required === undefined || p.required === 1 || prefix ? "" : "?";
```

And use `required: 0` on `overridingTypes.json`. I tried it and it works this time!

Reply:
That change doesn't appear to update the global variables:
```typescript
declare var ontouchcancel: ((this: Window, ev: TouchEvent) => any) | null;
declare var ontouchend: ((this: Window, ev: TouchEvent) => any) | null;
declare var ontouchmove: ((this: Window, ev: TouchEvent) => any) | null;
declare var ontouchstart: ((this: Window, ev: TouchEvent) => any) | null;
```
Those need to be `|undefined` as well I believe?
in Firefox if you do `>> ontouchcancel` you get
> ReferenceError: ontouchcancel is not defined

Reply:
Updated PR with your suggested changes in case that is desired behavior.

Reply:
Ah, you're right. We can modify it further:

```diff
             }
-            const requiredModifier = p.required === undefined || p.required === 1 ? "" : "?";
+            const required = p.required === undefined || p.required === 1;
+            const requiredModifier = required || prefix ? "" : "?";
             pType = p.nullable ? makeNullable(pType) : pType;
+            if (!required && prefix) {
+                pType += " | undefined"
+            }
```

Reply:
Updated.  I look forward to the day when someone git-blames me for this and I have no idea what any of it does.  😛